internal/server/handlers/k8s/database_cluster.go

```diff
@@ -406,3 +406,33 @@ func valueOrDefault(value *int, defaultValue int) int {
 	}
 	return *value
 }
+
+func (h *k8sHandler) CreateDatabaseClusterSecret(ctx context.Context,
+	namespace, dbName string,
+	engineType everestv1alpha1.EngineType,
+	secret *corev1.Secret,
+) (*corev1.Secret, error) {
+	secretCpy := secret.DeepCopy()
+	secretCpy.SetNamespace(namespace)
+	setDatabaseClusterSecretLabels(namespace, dbName, engineType, secretCpy)
+	return h.kubeConnector.CreateSecret(ctx, secretCpy)
+}
+
+func setDatabaseClusterSecretLabels(
+	namespace, dbName string,
+	engineType everestv1alpha1.EngineType,
+	secret *corev1.Secret,
+) {
+	labels := map[string]string{}
+	labels[common.DatabaseClusterNameLabel] = dbName
```

Comment:
btw, shall we set `ownerReference` on this secret in operator?

Reply:
> btw, shall we set ownerReference on this secret in operator?

I don't think that's needed, the DB operator generally ends up owning it and cleaning it up upon deletion. The only thing we're doing differently now is instead of letting the everest-operator create it by default, we create it

Reply:
Yes